distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso

```diff
@@ -49,6 +53,21 @@ type Array
     to_array : Array
     to_array self = @Builtin_Method "Array.to_array"
 
+    ## Checks whether this array is equal to `that`.
+
+       Arguments:
+       - that: The array to compare this array against.
+
+       Two arrays are considered equal, when they have the same length and
+       their items are pairwise equal.
+
+    == : Array -> Boolean
+    == self that =
+        if Meta.is_same_object Array self then Meta.is_same_object Array that else
```

Comment:
You can probably just check `Meta.is_same_object self that` instead of the 2-way check with `Array`. It will catch some actual arrays, but that doesn't matter, only makes the test faster sometimes.

Reply:
This is a different condition, I think. In case `self == Array`, I don't want to check the `else` branch.